pkg/netlink/netlink_retry_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netlink
+
+import (
+	"errors"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+)
+
+var _ = Describe("Netlink Retry Logic", func() {
+	Context("retryOnInterrupted", func() {
+		It("should succeed on first attempt when no error", func() {
+			callCount := 0
+			result, err := retryOnInterrupted(func() (string, error) {
+				callCount++
+				return "success", nil
+			})
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(Equal("success"))
+			Expect(callCount).To(Equal(1))
+		})
+
+		It("should retry on interrupted error and eventually succeed", func() {
+			callCount := 0
+			result, err := retryOnInterrupted(func() (string, error) {
+				callCount++
+				if callCount < MaxRetries-1 {
+					return "", netlink.ErrDumpInterrupted
+				}
+				return "success", nil
+			})
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(Equal("success"))
+			Expect(callCount).To(Equal(MaxRetries - 1))
+		})
+
+		It("should give up after maxRetries attempts", func() {
+			callCount := 0
+			result, err := retryOnInterrupted(func() (string, error) {
+				callCount++
+				return "", netlink.ErrDumpInterrupted
+			})
+
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(Equal(""))
+			Expect(callCount).To(Equal(MaxRetries))
+		})
+
+		It("should not retry on non-interrupt errors", func() {
+			callCount := 0
+			otherErr := errors.New("some other error")
+			result, err := retryOnInterrupted(func() (string, error) {
+				callCount++
+				return "", otherErr
+			})
+
+			Expect(err).To(Equal(otherErr))
+			Expect(result).To(Equal(""))
+			Expect(callCount).To(Equal(1))
+		})
+	})
+})
+
+func TestNetlinkRetry(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Netlink Retry Suite")
+}
```

Comment:
Let's move this to the standard _netlink_suite_test.go_ file as we may want to add unit tests for _adapter.go_ in the future.

```suggestion
func TestNetlink(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Netlink Suite")
}
```